src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java

```diff
@@ -434,64 +439,27 @@ public Mono<BookingResponseTO> updateBookingByReferenceCarrierBookingRequestRefe
     return bookingRepository
         .findByCarrierBookingRequestReference(carrierBookingRequestReference)
         .flatMap(checkUpdateBookingStatus)
+        .flatMap(booking -> {
+          // update the valid_until field for to be copied booking
+          // to ensure the copy is unique and the one to be used
+          booking.setValidUntil(OffsetDateTime.now());
+          return bookingRepository.save(booking).thenReturn(booking);
+        })
         .flatMap(
             b -> {
```

Comment:
I might be missing something here, but isn't the wrong booking being returned on line 457? We're returning the old booking with a non-null `validUntil` from line 446 which is later using when creating the deep objects on line 462. Shouldn't line 457 be `return bookingRepository.save(booking).thenReturn(booking)`?

Reply:
I think it's fine since the "b" on line 449 is not used, the other booking is created again from the bookingTO (line 451)

Reply:
but we are returning b on line 457 which contains the original id. We save the new booking, but then return the old booking which is then used in `createDeepObjectsForBooking` which uses the id from b.